identity-service/src/relay/ethTxRelay.js

```diff
@@ -0,0 +1,185 @@
+const EthereumWallet = require('ethereumjs-wallet')
+const EthereumTx = require('ethereumjs-tx')
+const axios = require('axios')
+const config = require('../config')
+const ethRelayerConfigs = config.get('ethRelayerWallets')
+const { ethWeb3 } = require('../web3')
+
+let ethRelayerWallets = [...ethRelayerConfigs] // will be array of { locked, publicKey, privateKey }
+ethRelayerWallets.forEach(wallet => {
+    wallet.locked = false
+})
+
+async function delay (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+// Calculates index into eth relayer addresses
+const getEthRelayerWalletIndex = (walletAddress) => {
+    let walletParsedInteger = parseInt(walletAddress, 16)
+    return walletParsedInteger % ethRelayerWallets.length
+}
+
+// Select from the list of eth relay wallet addresses
+// Return the public key that will be used to relay this address
+const queryEthRelayerWallet = (walletAddress) => {
+    return ethRelayerWallets[getEthRelayerWalletIndex(walletAddress)].publicKey
+}
+
+const getEthRelayerFunds = async (walletPublicKey) => {
+    return ethWeb3.eth.getBalance(walletPublicKey)
+}
+
+const selectEthWallet = async (walletPublicKey) => {
+  // TIMEOUT
+  let resolvedIndex = getEthRelayerWalletIndex(walletPublicKey)
+  while (ethRelayerWallets[index].locked) {
+    await delay(200)
+  }
+  // RIPE FOR MULTIPLE ENTRANCY ISSUES AND IS THREAD UNSAFE
+  ethRelayerWallets.locked = true
+  return ethRelayerWallets[resolvedIndex]
+}
+
+// Relay a transaction to the ethereum network
+const sendEthTransaction = async (req, txProps, reqBodySHA) => {
+    const {
+      contractAddress,
+      encodedABI,
+      senderAddress,
+      gasLimit
+    } = txProps
+
+    // Calculate relayer from  senderAddressjo
+    let selectedEthRelayerWallet = await selectEthWallet(senderAddress)
+    req.logger.info(`L1 txRelay - selected relayerIndex=${relayerIndex}, relayerPublicWallet=${selectedEthRelayerWallet.publicKey}`)
+    let ethGasPriceInfo = await getProdGasInfo(req.app.get('redis'), req.logger)
+
+    // Select the 'fast' gas price
+    let ethRelayGasPrice = ethGasPriceInfo.fastGweiHex
+    let txReceipt = await createAndSendTransaction(
```

Comment:
this should go into a try/catch/finally and the finally should unlock the wallet even if it fails